### PR TITLE
Remove cotire build

### PR DESCRIPTION
**Pull request by StormBytePP**
While compiling with CLang and libc++ cotire fails to build project, I adapted the original fork to make use of the original CMake FindProtobuf.cmake script, and compilation is now successful

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -50,11 +47,11 @@ option(ENABLE_CYCLIC_CHECK
 )
 option(ENABLE_TEST
   "Enable the tests"
-  ON
+  OFF
```
Comment (dschmidt) [inline CMakeLists.txt]: Why?

File: CMakeLists.txt [hunk 2]
```diff
@@ -50,11 +47,11 @@ option(ENABLE_CYCLIC_CHECK
...
 )
 option(ENABLE_DEMO
   "Enable building demo applications"
-  ON
+  OFF
```
Comment (dschmidt) [inline CMakeLists.txt]: Why?

File: CMakeLists.txt [hunk 3]
```diff
@@ -87,6 +84,10 @@ set(lucene_boost_libs
 include(Lucene++Docs)
 include(TestCXXAcceptsFlag)
 include(GNUInstallDirs)
+include(ExternalProject)
+
+set(PROTOBUF_HOME "/usr/local/Cellar/protobuf/3.6.0")
```
Comment (dschmidt) [inline CMakeLists.txt]: Why would you hardcode this path here?

Comment (alanw) [reply CMakeLists.txt]: This looks particularly dangerous.

**Pull request closed without merging**